tools/manifest/test262.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import print_function
+
+from typing import Dict, Optional, Text, Tuple, Any, Callable
+
+import re
+
+# Matches trailing whitespace and any following blank lines.
+_BLANK_LINES = r"([ \t]*[\r\n]{1,2})*"
+
+# Matches the YAML frontmatter block.
+_YAML_PATTERN = re.compile(r"/\*---(.*)---\*/" + _BLANK_LINES, re.DOTALL)
+
+_STRIP_CONTROL_CHARS = re.compile(r'[\x7f-\x9f]')
+
+
+class TestRecord:
+    @staticmethod
+    def _yaml_attr_parser(test_record: Dict[Text, Any], attrs: Text, name: Text, onerror: Callable[[Text], Any]) -> None:
+        import yaml
+        parsed = yaml.safe_load(re.sub(_STRIP_CONTROL_CHARS, ' ', attrs))
+        if parsed is None:
+            onerror("Failed to parse yaml in name %s" % name)
+            return
+
+        for key in parsed:
+            value = parsed[key]
+            if key == "info":
+                key = "commentary"
+            test_record[key] = value
+
+        if 'flags' in test_record:
+            for flag in test_record['flags']:
+                test_record[flag] = ""
+
+    @staticmethod
+    def _find_attrs(src: Text) -> Tuple[Optional[Text], Optional[Text]]:
+        match = _YAML_PATTERN.search(src)
+        if not match:
+            return (None, None)
+
+        return (match.group(0), match.group(1).strip())
+
+    @staticmethod
+    def parse(src: Text, name: Text, onerror: Callable[[Text], Any] = print) -> Optional[Dict[Text, Any]]:
```

Comment:
We never seem to set the `onerror` callable. I think the more normal thing in wptrunner would be to pass in a `logger` object as the first argument`.